File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeTwoLists(ListNode l1, ListNode l2) {
        if(l1==null){
            return l2;
        }
        else if(l2==null){
            return l1;
        }
        ListNode a,b=new ListNode(-1);
        a=b;
        while(l1!=null && l2!=null){
            if(l1!=null && l1.val<=l2.val){
                b.next=new ListNode(l1.val);
                l1=l1.next;
                b=b.next;
            }
            else if(l2!=null && l1.val>l2.val){
                b.next=new ListNode(l2.val);
                l2=l2.next;
                b=b.next;
            }
            
        }
        while(l1!=null){
            b.next=new ListNode(l1.val);
                l1=l1.next;
                b=b.next;
        }
        while(l2!=null){
            b.next=new ListNode(l2.val);
                l2=l2.next;
                b=b.next;
        }
        return a.next;
    }
}

File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeTwoLists(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        d=t=ListNode(0)
        while(l1!=None and l2!=None):
            if(l1.val>l2.val):
                t.next=ListNode(l2.val)
                l2=l2.next
            else:
                t.next=ListNode(l1.val)
                l1=l1.next
            t=t.next
        t.next=l1 or l2
        return d.next